### PR TITLE
Generator kenzie

### PR DESCRIPTION
**Pull request by m-dorsey**

### CODE REVIEW

File: nodes/ElseNode.java [hunk 1]
```diff
@@ -56,17 +56,40 @@ public void setHasIfParent(Boolean hasIfParent) {
...
     }
 
     @Override
     public String convertToPython() {
-        return null;
+        if (this.bodyNode == null) {
+            return "";
+        } else {
+            StringBuilder str = new StringBuilder();
+            str.append("else:");
```
Comment (trl6895) [inline nodes/ElseNode.java:93]: python needs newline and tab?

File: nodes/ElseifNode.java [hunk 2]
```diff
@@ -47,25 +47,42 @@ public String convertToJott() {
...
+        return str.toString();
     }
 
     @Override
     public String convertToPython() {
-        return null;
+        StringBuilder str = new StringBuilder();
+        str.append("elif ");
+        str.append(this.exprNode.convertToPython());
+        str.append(": ");
```
Comment (trl6895) [inline nodes/ElseifNode.java:90]: newline + tab?

File: nodes/WhileNode.java [hunk 3]
```diff
@@ -28,25 +28,42 @@ public String convertToJott() {
...
+        return str.toString();
     }
 
     @Override
     public String convertToPython() {
-        return null;
+        StringBuilder str = new StringBuilder();
+        str.append("while ");
+        str.append(this.exprNode.convertToPython());
+        str.append(": ");
```
Comment (trl6895) [inline nodes/WhileNode.java:71]: newline + tab?

**Review by trl6895 — APPROVED** [review]
lgtm

**Review by uvu1688 — APPROVED** [review]

**Pull request merged**